### ISSUE TITLE
How to implement transitions similar to KRPano

### ISSUE THREAD

**Issue by campmdabt** [google/marzipano#236]
We're migrating from KRPano to Marzipano since it's soooo much easier to develop on top of. Thanks for the hard work.

We've currently got a [complex transition](http://tours.atlasbayvr.com/arc/broadstonequeencity/CDR/?link=marzipano&startscene=scene_unit_a1_-_bath_scheme_1_shower&startactions=lookat(-179.68,5.36,115,0,0);) between different design options for some of our tours (use the option buttons on the left to switch).

Any ideas on how to implement here? We're currently using the [KRPano blend transition animation](https://krpano.com/docu/actions/#loadpano.blend) `SLIDEBLEND(0.7, 145.0, 0.1, linear)`, but I'm afraid Marzipano isn't really setup for that kind of transition out of the box.

**Comment by tjgq** [comment]
Have you taken a look at the transitions demo? ([code](https://github.com/google/marzipano/blob/master/demos/transitions/index.js), [live version](http://www.marzipano.net/demos/transitions/index.html))

In general, any transition that combines alpha blending with a viewport scaling or translation can be implemented as shown there.

**Comment by campmdabt** [comment]
Yes sir, I've taken a look at the demo and the code. The specific issues I
don't see a solution for how to implement would be a diagonal transition
line instead of a vertical or horizontal line, and a thin area of opacity
fade along that line rather than the entire layer.

**Comment by jordyno** [comment]
I second this, would be nice to have layer transitions at an angle with blurred edge of the transitioning layer. I am curious how we could achieve similar effect.

**Comment by tjgq** [comment]
Transitions are implemented through a combination of primitives. Currently there are primitives for opacity, translation/scaling (aka rect), and a general color matrix transform. This effect would require a new primitive that sets the opacity as a function of the x/y screen coordinates.

**Comment by JhianSV** [comment]
Any idea nowadays?
